.github/workflows/go.yml

```diff
@@ -79,6 +79,45 @@ jobs:
         with:
           fail_ci_if_error: true
 
+  test-integration:
+    name: Integration Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "1.15"
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Install binary
+        run: make install
+
+      - name: Use the binary
+        env:
+          AXM_ACCESS_TOKEN: ${{ secrets.TESTING_AZURE_1_STAGING_ACCESS_TOKEN }}
+          AXM_DEPLOYMENT_URL: ${{ secrets.TESTING_AZURE_1_STAGING_DEPLOYMENT_URL }}
+        run: |
+          axiom version -I
+          axiom dataset create -n=cli-test-$GITHUB_RUN_ID -d="CLI Integration test $GITHUB_RUN_ID"
+          cat logs.csv | axiom ingest cli-test-$GITHUB_RUN_ID --timestamp-format="2/Jan/2006:15:04:05 +0000" --timestamp-field="time"
+          cat logs.json | axiom ingest cli-test-$GITHUB_RUN_ID --timestamp-field="time"
+          cat logs.ndjson | axiom ingest cli-test-$GITHUB_RUN_ID
+          axiom ingest cli-test-$GITHUB_RUN_ID -f=logs.ndjson
+          axiom ingest cli-test-$GITHUB_RUN_ID -f=logs.json -f=logs.csv
+          axiom dataset info cli-test-$GITHUB_RUN_ID
+          axiom dataset list
+          axiom dataset delete cli-test-$GITHUB_RUN_ID
```

Comment:
If the test fails we have a dataset that's not getting deleted. I don't know if actions can do a post_step or something but might be worth looking into

Reply:
Good catch, I think [this](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions) will do the trick.